gnat2goto/driver/tree_walk.adb

```diff
@@ -3512,6 +3555,21 @@ package body Tree_Walk is
       return Typecast_If_Necessary (Mod_Ret, Ret_Type, Global_Symbol_Table);
    end Do_Operator_Mod;
 
+   function Do_Operator_Mod (N : Node_Id) return Irep is
+      Impossible_Exception : exception;
+      Op_Kind : constant Irep_Kind :=
+        (case N_Op (Nkind (N)) is
+           when N_Op_Add => I_Op_Add,
+           when N_Op_Multiply => I_Op_Mul,
+           when others => raise Impossible_Exception
+                          with "this case is excluded by the precondition");
+      Lhs : constant Irep := Do_Expression (Left_Opnd (N));
+      Rhs : constant Irep := Do_Expression (Right_Opnd (N));
+      Expr_Type : constant Irep := Do_Type_Reference (Etype (N));
+   begin
+      return Do_Operator_Mod (Lhs, Op_Kind, Rhs, Expr_Type);
```

Comment:
FWIW I wanted to remove this version of `Do_Operator_Mod` entirely, but it's used in codegen as well. I think we should probably rename it though, as it doesn't really the normal `Do_XXX` scheme of functions (i.e. it doesn't actually do any parsing of node ids, it just generates code for mod operations).